pkgs/apps/psi4/default.nix

```diff
@@ -36,9 +36,40 @@ let
     '';
   });
 
-  libintSrc = fetchurl {
-    url = "https://github.com/loriab/libint/releases/download/v0.1/Libint2-export-7-7-4-7-7-5_1.tgz";
-    sha256 = "16vgnjpspairzm72ffjc8qb1qz0vwbx1cq237mc3c79qvqlw2zmn";
+  libint = stdenv.mkDerivation rec {
+    pname = "libint";
+    version = "0.1";
+
+    src = fetchurl {
+      url = "https://github.com/loriab/${pname}/releases/download/v${version}/Libint2-export-7-7-4-7-7-5_1.tgz";
+      hash = "sha256-tn7BKd44HTZYPUNgFvriG3wcFkZMOidO/Tmqq6+0b5s=";
+    };
+
+    nativeBuildInputs = [ cmake ];
+    propagatedBuildInputs = [ eigen boost mpfr ];
+
+    preConfigure = ''
+      ulimit -s 65536
+    '';
+
+    cmakeFlags = specialInstallCmakeFlags ++ [
+      "-DLIBINT2_SHGAUSS_ORDERING=gaussian"
+      "-DCMAKE_BUILD_TYPE=Release"
+      "-DBUILD_SHARED=ON"
+      "-DBUILD_STATIC=ON"
+      "-DMERGE_LIBDERIV_INCLUDEDIR=OFF"
+      "-DBUILD_FPIC=ON"
+      "-DENABLE_GENERIC=ON"
+      "-DCMAKE_CXX_FLAGS=-fPIC"
+      "-DENABLE_CXX11API=ON"
+      "-DCMAKECONFIG_INSTALL_DIR=share/cmake/Libint2"
+    ];
+
+    # CMake generates a directory '$out' (not the expanded variable but literally) in the build
+    # directory. Within, the share/cmake files are, and they need to be copied manually now.
+    postInstall = ''
+      cp -r \$out/share $out/.
```

Comment:
You can avoid this by using `cmakeFlagsArray` instead of `cmakeFlags`. This should allow for proper shell expansion.

Reply:
Unfortunately not in this case. It's a bug in this CMake system, not a problem of the shell expansion.

Reply:
Hm, i will try it nevertheless. Now I am not so sure anymore. :open_mouth:

Reply:
If  you write `cmakeFlagsArray+=(-DCMAKE_INSTALL_PREFIX=$out)` in the `preConfigure` phase, then you get shell substitution and cmake will never see the literal "$out", but the actual path "/nix/store/...." (?) Am I overlooking something here?

Reply:
Ah no, probably you are right. I am possibly introducing this indirectly via the `specialInstallCmakeFlags` earlier. I will wait for the build to finish (some hours thanks to libint). If it works, I will push the changes.